table/Table.jsx

```diff
@@ -969,38 +1084,72 @@ function AggregateRow<T>({
         className={className}
         style={style}
       >
-        {leftPinnedColumns.length ? (
-          <div className={css(styles.leftPinnedDrawer)}>
-            {leftPinnedColumns.map(({id, AggregateCell}) => (
-              <AggregateCell key={id} rows={data} />
-            ))}
-          </div>
-        ) : null}
+        <PinnedColumns
+          columns={leftPinnedColumns}
+          scrollContext={scrollContext}
+          type="left"
+        >
+          {({id, AggregateCell}) => <AggregateCell key={id} rows={data} />}
+        </PinnedColumns>
         <div className={css(styles.unpinnedDrawer)}>
           {unpinnedColumns.map(({id, AggregateCell}) => (
             <AggregateCell key={id} rows={data} />
           ))}
         </div>
-        {rightPinnedColumns.length ? (
-          <div className={css(styles.rightPinnedDrawer)}>
-            {rightPinnedColumns.map(({id, AggregateCell}) => (
-              <AggregateCell key={id} rows={data} />
-            ))}
-          </div>
-        ) : null}
+        <PinnedColumns
+          columns={rightPinnedColumns}
+          scrollContext={scrollContext}
+          type="right"
+        >
+          {({id, AggregateCell}) => <AggregateCell key={id} rows={data} />}
+        </PinnedColumns>
       </div>
     </Clickable>
   );
 }
 
+function PinnedColumns<T>({
+  columns,
+  children,
+  scrollContext,
+  type,
+}: {
+  +columns: Array<Column<T>>,
+  +children: (Column<T>) => React.Node,
+  +scrollContext: *,
+  +type: "left" | "right",
+}) {
+  return columns.length ? (
```

Comment:
Because `scrollContext` is a context, we can remove it from the props and access it via `useContext`

```suggestion
  const scrollContext = React.useContext(TableScrollContext);
  return columns.length ? (
```